pyresample/bucket/__init__.py

```diff
@@ -208,27 +237,18 @@ def _call_pandas_groupby_statistics(self, scipy_method, data, fill_value=None, s
         # Calculate the min of the data falling to each bin
         out_size = self.target_area.size
 
-        # merge into one Dataframe
-        df = dd.concat([dd.from_dask_array(self.idxs), dd.from_dask_array(weights)],
-                       axis=1)
-        df.columns = ['x', 'values']
-
-        if scipy_method == 'min':
-            statistics = df.map_partitions(lambda part: part.groupby(
-                                           np.digitize(part.x,
-                                                       bins=np.linspace(0, out_size, out_size)
-                                                       )
-                                           )['values'].min())
+        # sort idxs and weights
+        order = da.from_delayed(self._sort_weights(statistic_method, weights),
```

Comment:
I would move `_sort_weights` outside of the class. Dask is likely having to serialize the resampler object when this function is called. You might also want to look at setting `pure=True` when you create the delayed version of `_sort_weights`. I forget if that is the default for a delayed function.